lib/jekyll/readers/data_reader.rb

```diff
@@ -3,73 +3,28 @@
 module Jekyll
   class DataReader
     attr_reader :site, :content
+
+    EXTENSIONS = %w(.yaml .yml .json .csv).freeze
+
     def initialize(site)
       @site = site
-      @content = {}
       @entry_filter = EntryFilter.new(site)
     end
 
-    # Read all the files in <dir> and adds them to @content
-    #
-    # dir - The String relative path of the directory to read.
-    #
-    # Returns @content, a Hash of the .yaml, .yml,
-    # .json, and .csv files in the base directory
-    def read(dir)
-      base = site.in_source_dir(dir)
-      read_data_to(base, @content)
-      @content
-    end
-
-    # Read and parse all .yaml, .yml, .json, .csv and .tsv
-    # files under <dir> and add them to the <data> variable.
-    #
-    # dir - The string absolute path of the directory to read.
-    # data - The variable to which data will be added.
-    #
-    # Returns nothing
-    def read_data_to(dir, data)
-      return unless File.directory?(dir) && !@entry_filter.symlink?(dir)
+    def read_data(dir)
+      site.reader.get_entries(dir, site.config["data_dir"]).map do |entry|
+        next unless EXTENSIONS.include?(File.extname(entry))
 
-      entries = Dir.chdir(dir) do
-        Dir["*.{yaml,yml,json,csv,tsv}"] + Dir["*"].select { |fn| File.directory?(fn) }
-      end
-
-      entries.each do |entry|
-        path = @site.in_source_dir(dir, entry)
+        path = @site.in_source_dir(File.join(dir, site.config["data_dir"], entry))
         next if @entry_filter.symlink?(path)
 
-        if File.directory?(path)
-          read_data_to(path, data[sanitize_filename(entry)] = {})
-        else
-          key = sanitize_filename(File.basename(entry, ".*"))
```

Comment:
@pathawks After some testing locally, I believe the translation of this to collections is needed. The key used to be generated here and used in the line below, but with the collection handling it, we do not have a key being set. So if I want data to exist I need to have something like 
```yaml
---
products:
- name: sugar
  price: 5.3
- name: salt
  price: 2.5
```

in `_data/products.yaml` in order for `site.data.products` to be properly initiliazed, and then it still would be a bit different than how it is now I believe. Give it a try and have a index.html with "{{site.data | jsonify}}" in it. Take a look at the output compared to how it is currently without this Branch/PR

Reply:
@pathawks to see what I’m saying easily, try making a new site with `_data/[anything here is fine].yml` With it being a valid yaml data set like that in the data.feature definitions.

Make an index.html that contains {{site.data | jsonify }} and {{ site.collections | jsonify }}

You will see the difference I’m seeing where the collections appears to get the data but the site.data hash only seems to get the name of the data files as a key but no values.